code/datums/components/obj_projectile_damage.dm

```diff
@@ -0,0 +1,32 @@
+/datum/component/obj_projectile_damage
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+
+/datum/component/obj_projectile_damage/Initialize()
+	if(!istype(parent, /obj))
+		return COMPONENT_INCOMPATIBLE
+	RegisterSignal(parent, COMSIG_ATOM_HITBY_PROJ, .proc/projectile_collide)
+	RegisterSignal(parent, COMSIG_ATOM_EXAMINE, .proc/examine)
+
+/datum/component/obj_projectile_damage/proc/projectile_collide(owner, var/obj/projectile/P)
+	var/obj/O = parent
+	if(P.proj_data.damage_type & (D_KINETIC | D_ENERGY | D_SLASHING))
+		if (O && (O._health/O._max_health) <= 0.5 && prob(abs((O._health/O._max_health) - 1) * 60))
+			var/obj/decal/cleanable/machine_debris/gib = null
+			gib = make_cleanable(/obj/decal/cleanable/machine_debris, O.loc)
```

Comment:
```suggestion
			var/obj/decal/cleanable/machine_debris/gib = make_cleanable(/obj/decal/cleanable/machine_debris, O.loc)
```
Assuming this doesn't violate some line length issue, seems neater to write as one.